packages/terminal/src/browser/base/terminal-service.ts

```diff
@@ -33,6 +33,11 @@ export interface TerminalService {
 
     readonly all: TerminalWidget[];
 
+    /**
+     * All terminal widgets that aren't hidden (see `TerminalWidget.hidden`).
+     */
+    readonly allVisible: TerminalWidget[];
```

Comment:
@planger I'm not a huge fan of adding this new property when we already have `all` and it's used once, wdyt?

Reply:
Right, I haven't addressed the issue with the name `hidden` yet. I suggest to flesh out this name a bit, maybe `hiddenFromUser`?
Regarding the `allVisible` property: My thinking was that if ever someone needed to show a list of terminals in the UI, they should show the list without those `hiddenFromUser`. Thus I preemptively added a property for that to avoid forgetting about that in the future. However, I agree it might be unnecessary for now, as we only use this property once at the moment.

I'll update the change with the rename and remove the new property and do the filtering in the frontend contribution instead. Please let me know what you think.